src/linked_list.py

```diff
@@ -0,0 +1,59 @@
+class Node:
+    def __init__(self, val, next_node):
+        self.val = val
+        self.next_node = next_node
+
+    def __repr__(self):
+        return f"({self.val})->{self.next_node}"
+
+
+class LinkedList:
+    def __init__(self, iterable=None):
+        self.head = None
+        self._length = 0
+        if type(iterable) in [list, tuple, str]:
+            for item in iterable:
+                self.push(item)
+        elif iterable is not None:
+            raise TypeError("An iterable must be provided.")
```

Comment:
This is a confusing error message, because from the user's perspective an iterable DOESN'T need to be provided. They can just use the default `None`, or even pass in `iterable=None`.